packages/assets-controllers/src/NftController.test.ts

```diff
@@ -1955,6 +1964,147 @@ describe('NftController', () => {
         tokenURI: '',
       });
     });
+
+    it('should, when passed a networkClientId, add an NFT with the correct chainId and metadata fetched using networkClientId', async () => {
+      nock('https://testtokenuri-1.com')
+        .get('/')
+        .reply(
+          200,
+          JSON.stringify({
+            image: 'test-image-1',
+            name: 'test-name-1',
+            description: 'test-description-1',
+          }),
+        );
+
+      nock('https://testtokenuri-2.com')
+        .get('/')
+        .reply(
+          200,
+          JSON.stringify({
+            image: 'test-image-2',
+            name: 'test-name-2',
+            description: 'test-description-2',
+          }),
+        );
+
+      nock('https://testtokenuri-3.com')
+        .get('/')
+        .reply(
+          200,
+          JSON.stringify({
+            image: 'test-image-3',
+            name: 'test-name-3',
+            description: 'test-description-3',
+          }),
+        );
+
+      const { nftController, getNetworkClientByIdSpy } = setupController({
+        getERC721TokenURIStub: jest.fn().mockImplementation((tokenAddress) => {
+          switch (tokenAddress) {
+            case '0x01':
+              return 'https://testtokenuri-1.com';
+            case '0x02':
+              return 'https://testtokenuri-2.com';
+            default:
+              throw new Error('Not an ERC721 token');
+          }
+        }),
+        getERC1155TokenURIStub: jest.fn().mockImplementation((tokenAddress) => {
+          switch (tokenAddress) {
+            case '0x03':
+              return 'https://testtokenuri-3.com';
+            default:
+              throw new Error('Not an ERC1155 token');
+          }
+        }),
+      });
+
+      getNetworkClientByIdSpy.mockImplementation((networkClientId) => {
+        switch (networkClientId) {
+          case 'sepolia':
+            return {
+              configuration: {
+                chainId: SEPOLIA.chainId,
+              },
+            };
+          case 'goerli':
+            return {
+              configuration: {
+                chainId: GOERLI.chainId,
+              },
+            };
+          case 'customNetworkClientId-1':
+            return {
+              configuration: {
+                chainId: '0xa',
+              },
+            };
+          default:
+            return {
+              configuration: {
+                chainId: '0x1',
+              },
+            };
```

Comment:
normal getNetworkClientById behavior is to throw if it's missing

Reply:
done here: https://github.com/MetaMask/core/pull/1698/commits/5fd50ec7deb2b8e06f3221c78abd7363df0d7039